core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -529,7 +529,6 @@ public List<QualifiedObjectName> listTables(Session session, QualifiedTablePrefi
                 metadata.listTables(connectorSession, prefix.getSchemaName()).stream()
                         .map(convertFromSchemaTableName(prefix.getCatalogName()))
                         .filter(table -> !isExternalInformationSchema(catalogHandle, table.getSchemaName()))
-                        .filter(prefix::matches)
```

Comment:
Without this filter, this method may return unmatched tables if the connectors include such objects. I have a small concern to trust the connectors' implementation.

Reply:
Sorry, i didn't refresh the page before merging and didn't see the comment.

Such a behavior would be a bug in the connector and such be fixed there.
if we wanted to allow connector to return unfiltered results, that should be documented (and tested).
BTW, we didn't have `prefix.matches` calls (or equivalent) in eg listTableColumns.